router/pkg/config/config.go

```diff
@@ -230,9 +230,12 @@ type EngineExecutionConfiguration struct {
 }
 
 type SecurityConfiguration struct {
-	BlockMutations              bool `yaml:"block_mutations" envDefault:"false" env:"SECURITY_BLOCK_MUTATIONS"`
-	BlockSubscriptions          bool `yaml:"block_subscriptions" envDefault:"false" env:"SECURITY_BLOCK_SUBSCRIPTIONS"`
-	BlockNonPersistedOperations bool `yaml:"block_non_persisted_operations" envDefault:"false" env:"SECURITY_BLOCK_NON_PERSISTED_OPERATIONS"`
+	BlockMutations                       bool  `yaml:"block_mutations" envDefault:"false" env:"SECURITY_BLOCK_MUTATIONS"`
+	BlockSubscriptions                   bool  `yaml:"block_subscriptions" envDefault:"false" env:"SECURITY_BLOCK_SUBSCRIPTIONS"`
+	BlockNonPersistedOperations          bool  `yaml:"block_non_persisted_operations" envDefault:"false" env:"SECURITY_BLOCK_NON_PERSISTED_OPERATIONS"`
+	MaxQueryDepth                        int   `yaml:"max_query_depth,omitempty" envDefault:"0" env:"SECURITY_MAX_QUERY_DEPTH"`
+	QueryDepthCacheSize                  int64 `yaml:"query_depth_cache_size,omitempty" envDefault:"1024" env:"SECURITY_QUERY_DEPTH_CACHE_SIZE"`
+	DisableDepthLimitPersistedOperations bool  `yaml:"disable_depth_limit_persisted_operations,omitempty" envDefault:"false" env:"SECURITY_DISABLE_DEPTH_LIMIT_PERSISTED_OPERATIONS"`
```

Comment:
I'm not set on this config value name, it seems both long and a bit confusing; I'd welcome a clearer alternative!

Reply:
I agree, good call. What about? This also gives room for extensibility.

```yaml
depth_limit:
   enabled: true
   limit: 7
   persisted_operations: false
```

Reply:
@jensneuse  
Makes sense to group all of the configuration once there are a few elements. Should we have an `enabled` flag at all? I was previously just using `limit` to signify if it's enabled, if it's set to a value about 0. 

I'd also have the cache size in that configuration

Reply:
We use `enabled` everywhere to give the developer an option to disable parts of the config without removing the code.

Reply:
Yeah, we kinda made it a pattern to have an object with "enabled".